### ISSUE TITLE
about training details.

### ISSUE THREAD

**Issue by jinxxo-j** [ronferens/hyperpose#2]
Hello! Your research is fascinating. While reproducing your work, I noticed a large performance gap in the chess sequence of the 7Scenes dataset. After checking the supplementary material, I confirmed that you used a 3-stage training pipeline.

In the JSON file, I found that during the second-stage training, you used:

`"freeze": true,
"freeze_exclude_phrase": [
  "hyper_in_t_proj",
  "hyper_in_t",
  "hyper_in_t_fc_2",
  "hypernet_t_fc_h2"
]`

and during the third-stage training, you used:
`"freeze": true,
"freeze_exclude_phrase": [
  "hyper_in_rot_proj",
  "hyper_in_rot",
  "hyper_in_rot_fc_2",
  "hypernet_rot_fc_h2"
]
`
I am trying to follow this setup for training.
However, in your paper you mention:

> “For single-scene models, the learning rate is reduced by 25% every 20 epochs for indoor localization and every 200 epochs for outdoor localization, with maxima of 100 and 600 epochs, respectively.”

Does this mean that the second and third training stages are not limited to only 10 epochs each?
Could you please clarify how many epochs were used for each stage of training?

**Comment by ronferens** [comment]
Hi @jinxxo-j ,
Thanks for your kind works, I'm very happy to hear that you find our work interesting.

To be honest, it has been a while since I've perfomed the models' training, but I'll do my best to assist you.
As a starting point, if you haven't still done that, I would recommend running the pre-trained model I've shared within your local testing environment. This will ensure that everything works well and you are ready for reproducing the paper's results.
The pretrained model can be found here - [pre-trained model .zip](https://www.dropbox.com/scl/fi/bj72wz17u77phidvtgyem/drive-download-20241114T145344Z-001.zip?rlkey=x0gzacfep9fe9kgq2yxc3tfq0&e=2&st=icbxe3s7&dl=0)

I've reviewed my Chess scene model training notes:
- **First training stage** this is what I've logged - **lr e-4, 20/100/0.75, 256/512, bs8, 0/0, no jitter.** This means, that the model was trained for 100 epoches, after every 20 epochs the learning rate dropped by 25%, the batch size was 8, both Sx and Sq were set to 0 and I've disabled the jitter augmentation. The hyper_dim_t and hyper_dim_rot where set to 256 and 512 respectively.
- **Second stage** - X fine-tunning - ran for 4 epochs with learning rate of 1e-3
- **Third stage** - Q fine-tunning - ran for 7 epochs with learning rate of 1e-3
For both stages II and III, I've applied the freeze configuration as you have correctly described.

As for the number of epochs in the second and third stages, I've checked my notes and for some scenes in the 7Scenes dataset, I did log fine-tunning stages that ran beyond 10 epochs.

I truly hope this helps you,
Ron
